app/code/Magento/Catalog/Model/Indexer/Category/Product/AbstractAction.php

```diff
@@ -343,6 +334,19 @@ protected function getNonAnchorCategoriesSelect(Store $store)
                 ]
             );
 
+            if ($store->getId() != 0) {
+                $select->joinInner(
+                    ['cpw' => $this->getTable('catalog_product_website')],
+                    'cpw.product_id = ccp.product_id',
+                    []
+                )->where(
+                    'cpw.website_id = ?',
+                    $store->getWebsiteId()
+                )->where(
+                    'cc.path LIKE ' . $this->connection->quote($rootPath . '/%')
```

Comment:
my suggestion is to use regex instead like, to improve performance of the query.

Reply:
I thought like was faster than regex ?!